editor/components/table-of-contents/panel.js

```diff
@@ -58,8 +56,20 @@ function TableOfContentsPanel( { blocks } ) {
 	);
 }
 
+const blockCount = ( blocksArray = [], name ) => reduce(
```

Comment:
Should this be a propers elector?

Reply:
Yes, it is a good idea and has some advantages. Namely, we can make use of the state.editor.present.blocksByUid structure which is flat so we can have a much simpler and efficient algorithm when compared with the previous recursive version.